docs/tutorials/text_prediction/automm.md

```diff
@@ -0,0 +1,393 @@
+
+`env` contains the environment/machine related configurations. For example, the optimal values of `per_gpu_batch_size` and `per_gpu_batch_size_evaluation` are closely related to the GPU memory size.
+
+You can flexibly customize any hyper-parameter in `config` via the `hyperparameters` argument of `.fit()`. To access one hyper-parameter in `config`, you need to traverse from top-level keys to bottom-level keys and join them together with `.` For example, if you want to change the per gpu batch size to 16, you can set `hyperparameters={"env.per_gpu_batch_size": 16}`.
+
+
+Next, we will introduce how to customize the default config to train different models.
+
+## Image Prediction
+
+Let's first train an image classifier on the PetFinder dataset. AutoMM supports loading diverse [timm](https://github.com/rwightman/pytorch-image-models) backbones. We can choose one image backbone, e.g., [swin_tiny_patch4_window7_224](https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/swin_transformer.py#L69), by using hyper-parameter `model.timm_image.checkpoint_name`.
+
+
+```{.python .input}
+from autogluon.text.automm import AutoMMPredictor
+predictor = AutoMMPredictor(label=label_col)
+predictor.fit(
+    train_data=train_data,
+    config=config,
+    hyperparameters={
+        "model.names": ["timm_image"],
+        "model.timm_image.checkpoint_name": "swin_tiny_patch4_window7_224",
+    },
+    time_limit=60, # seconds
+)
+```
+
+During `.fit()`, `train_data` is automatically divided into training and validation sets. AutoMM uses the validation score to select the top k (default 3) model checkpoints and average them as the final model. When calling `.fit()` you can also set `tuning_data=validation_data` to use your specified `validation_data`.
+
+Here AutoMM uses only image data and automatically ignores other data since `model.names` only include `timm_image`.
+
+Then we can evaluate the trained model on test data. AutoMM supports evaluation with multiple metrics.
+
+
+```{.python .input}
+scores = predictor.evaluate(test_data, metrics=["accuracy", "quadratic_kappa"])
+print(scores)
+```
+
+Here we use a tiny backbone and a tight time budget for a quick demonstration. To perform well on your tasks, you may choose a larger backbone and do not set a time limit.
+
+Given data without ground truth labels, AutoMM can make predictions.
+
+
+```{.python .input}
+predictions = predictor.predict(test_data.drop(columns=label_col))
+predictions[:5]
+```
+
+For classification tasks, we can get the probabilities of all classes.
+
+
+```{.python .input}
+probas = predictor.predict_proba(test_data.drop(columns=label_col))
+probas[:5]
+```
+
+Note that calling `.predict_proba` on one regression task will throw an exception.
+
+Extract embeddings can be easily done via `.extract_embedding()`
+
+
+```{.python .input}
+embeddings = predictor.extract_embedding(test_data.drop(columns=label_col))
+embeddings[0]
```

Comment:
May be we can just output the shape of the embedding. Printing the embedding values won't make it more clear.

Reply:
Sure.